rpc/src/v1/helpers/light_fetch.rs

```diff
@@ -367,8 +368,38 @@ impl LightFetch {
 							}
 							future::ok::<_,OnDemandError>(matches)
 						}) // and then collect them into a vector.
-						.map(|matches| matches.into_iter().map(|(_, v)| v).collect())
 						.map_err(errors::on_demand_error)
+						// retrieve transaction hash.
+						.and_then(move |matches| {
+							let mut blocks = BTreeMap::new();
+							let mut result: Vec<Log> = matches.into_iter().map(|(_, v)| {
+								{
+									let block_hash = v.block_hash.as_ref().expect("Previously initialized with value; qed");
+									blocks.entry(block_hash.clone()).or_insert_with(|| {
```

Comment:
`block_hash` is copy, so we don't need `clone`.

Reply:
Actually, I think this is `rpc::v1::types::hash` which is not `Copy` but we should refactor that to be consistent in the codebase!